synadm/cli/room.py

```diff
@@ -245,10 +246,13 @@ def members(helper, room_id):
     "--no-purge", is_flag=True, default=False, show_default=True,
     help="""Prevent removing of all traces of the room from your
     database.""")
+@click.option(
+    "--v1", is_flag=True, default=False, show_default=True,
+    help="""Use version 1 of the room delete API instead of version 2""")
```

Comment:
I like this idea very much. I always had some auto-magic in my mind that detects when Synapse is old enough to use v1, otherwise v2.

I think having it "selectable" by the user/admin like this is a much easier and less errorprone solution. In the end a Matrix admin should have the right to choose what they want to do. Too much magic is not good anyway.

Perfect solution!